pipenv/installers.py

```diff
@@ -56,12 +64,52 @@ def __init__(self, desc, c):
 
 
 class Installer(object):
+    __metaclass__ = ABCMeta
```

Comment:
This shows no obvious improvement comparing to the plain parent class and `NotImplementedError`, but I am OK with it.

Reply:
Sorry, force of habit. It's a very slight advantage (some IDEs and linters will automatically warn if you subclass and forget to implement unimplemented methods using ABCmeta), but I'm happy to back it out if you'd prefer.